k9mail/src/main/java/com/fsck/k9/activity/setup/AccountSettings.java

```diff
@@ -733,6 +749,28 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
         }
     }
 
+    private void showApgDeprecationDialog() {
+        ApgDeprecationWarningDialog fragment = ApgDeprecationWarningDialog.newInstance();
```

Comment:
That's because the dialog is dismissed, then recreated on orientation changes. And in the dismiss listener you show the preference dialog.
I changed it to so the action is only performed in the cancel listener.